### ISSUE TITLE
SVG output incorrect on devices with high DPI

### ISSUE THREAD

**Issue by jamcoupe** [szimek/signature_pad#219]
I built the library from master (as I wanted to use the new SVG output).

The SVG output on devices with a high DPI is incorrect.

I have provided a plunker and an image to demonstrate the issue I am seeing: https://plnkr.co/edit/o4mgY6i8H1RAOCWFWX2m?p=preview on a device with a high DPI draw in the red area, click on "Display as SVG Image" and see that the result is much smaller than expected.

![Image](https://s30.postimg.org/51rlqus4x/Screenshot_20170209_170854.png)

**Comment by kashban** [comment]
This happens with PNG too. Device is a Galaxy Tab S2, pixel ratio is 2.

**Comment by kashban** [comment]
http://stackoverflow.com/a/28048865/1888468

I implemented this answer in an Angular 1 directive to solve both scaling and high resolution devices:

```
function resizeCanvas() {
	            var ratio =  Math.max($window.devicePixelRatio || 1, 1);
	            if (canvas != null){
		            canvas.width = canvas.offsetWidth * ratio;
		            canvas.height = canvas.offsetHeight * ratio;
		            canvas.getContext("2d").scale(ratio, ratio);
		            // redraw image
		            if (model[idLocal] != null) drawImageData(model[idLocal], canvas, ratio);
		            else signaturePad.clear();
	            }
	        }
	        
	        angular.element($window).bind('resize', resizeCanvas);
	
	        signaturePad = new SignaturePad(canvas);

	        resizeCanvas();
        
	        // To keep aspect ration of the source image and fit it into the canvas
	        function drawImageData(imageData, canvas, pixelRatio){
	        	var img = new Image();
	            img.onload = function(){
	                var ct = document.getElementById('measure'); 
	                ct.appendChild(img);
	                var wrh = img.width / img.height;
	                var newWidth = canvas.width;
	                var newHeight = newWidth / wrh;
	                if (newHeight > canvas.height) {
	                    newHeight = canvas.height;
	                    newWidth = newHeight * wrh;
	                }
	                ct.removeChild(img);
	                canvas.getContext("2d").drawImage(img, 0, 0, newWidth / pixelRatio, newHeight / pixelRatio);
	            }
	            img.src = "data:image/png;base64," + imageData;
	        }

```

**Comment by szimek** [comment]
With SVG there seems to be an easy fix - just divide `maxX` and `maxY` (which are used in `viewBox` attribute) by `window.devicePixelRatio`. 

I'm not sure though how it relates to `width` and `height` attributes, which seem to be required by Firefox. I'll release a new beta version as soon as I figure it out.

**Comment by szimek** [comment]
@jamcoupe I've just released a new version, where this is hopefully fixed: signature_pad@1.6.0-beta.9. Could you check it out? If it works, I think I can finally release 1.6.0 stable ;)
